src/crab/type_to_num.cpp

```
@@ -77,14 +107,20 @@ void TypeToNumDomain::operator|=(TypeToNumDomain&& other) {
     if (other.is_bottom()) {
         return;
     }
+    this->ptr_sum_bindings = intersect_bindings(this->ptr_sum_bindings, other.ptr_sum_bindings);
     this->join_selective(other);
     this->types |= std::move(other.types);
 }
 
 TypeToNumDomain TypeToNumDomain::operator&(const TypeToNumDomain& other) const {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)